src/main/java/com/custom/fuelsdk/PaginationETSoapObject.java

```diff
@@ -0,0 +1,365 @@
+      //
+
+      for (String property : internalTypeAnnotation.unretrievable()) {
+        internalProperties.remove(property);
+      }
+    }
+
+    //
+    // Perform the SOAP retrieve:
+    //
+
+    //Soap soap = connection.getSoap();
+    Soap soap = null;
+
+    RetrieveRequest retrieveRequest = new RetrieveRequest();
+
+    if (continueRequest == null) {
+      // if soapObjectType is specified, use it; otherwise, default
+      // to the name of the internal class representing the object:
+      if (soapObjectName != null) {
+        retrieveRequest.setObjectType(soapObjectName);
+        soap = connection.getSoap("retrieve", soapObjectName);
+      } else {
+        retrieveRequest.setObjectType(internalType.getSimpleName());
+        soap = connection.getSoap("retrieve", internalType.getSimpleName());
+      }
+      retrieveRequest.getProperties().addAll(internalProperties);
+
+      if (expression.getOperator() != null) {
+        //
+        // Convert the property names to their internal counterparts:
+        //
+
+        String property = expression.getProperty();
+        if (property != null) {
+          expression.setProperty(getInternalProperty(type, property));
+        }
+        for (ETExpression subexpression : expression.getSubexpressions()) {
+          String p = subexpression.getProperty();
+          if (p != null) {
+            subexpression.setProperty(getInternalProperty(type, p));
+          }
+        }
+
+        retrieveRequest.setFilter(toFilterPart(expression));
+      }
+    } else {
+      if (continueRequest != null) {
+        retrieveRequest.setContinueRequest(continueRequest);
+      }
+    }
+
+//    //todo reduce batchsize
+//    RetrieveOptions retrieveOptions = new RetrieveOptions();
+////    retrieveOptions.setBatchSize(1);
+//    retrieveRequest.setOptions(retrieveOptions);
+
+    if (logger.isTraceEnabled()) {
+      logger.trace("RetrieveRequest:");
+      logger.trace("  objectType = " + retrieveRequest.getObjectType());
+      StringBuilder line = new StringBuilder("  properties = { ");
+      //String line = null;
+      for (String property : retrieveRequest.getProperties()) {
+
+        line.append(property).append(", ");
+      }
+      if (retrieveRequest.getProperties().size() > 0) {
+        line.setLength(line.length() - 2);
+                /*if (line == null) {
+                    line = "  properties = { " + property;
+                } else {
+                    line += ", " + property;
+                }*/
+      }
+      line.append(" }");
+      logger.trace(line.toString());
+      //logger.trace(line + " }");
+      if (filter != null) {
+        logger.trace("  filter = " + toFilterPart(expression));
+      }
+    }
+
+    logger.trace("calling soap.retrieve...");
+
+    RetrieveRequestMsg retrieveRequestMsg = new RetrieveRequestMsg();
+    retrieveRequestMsg.setRetrieveRequest(retrieveRequest);
+
+    //todo temp fix for continue request
```

Comment:
why a temp fix? Can you add a comment on whats the permanent fix, pls

Reply:
Removed the comment. This is the fix(temp) till FuelSDK team will provides a fix in an Offical release.